WISE-Community/WISE-Client
# feat(Question Bank): Click question to add to student response input

## Changes

- Allow students to click on questions in the question bank to add the question text into the response input. The student can then modify the question if they would like before they submit it.
- Allow authors to enable and disable this feature
- Allow authors to change the name of the question bank
- After a student uses a question, indicate that the question has been used
- Store which question was used in the student response
- When questions are authored, they should now have a unique id
  - New Question Bank content is marked as version 2 which contains question objects that have an id and text field
  - Old Question Bank content contains question strings

## Test

As a teacher
1. Create a new Peer Chat component
2. Enable the Question Bank in the Peer Chat
3. Set a Custom Label for the Question Bank
4. Enable Student Click To Add
5. Set a Reference Component
6. Set Grouping Logic
7. Add some Question Bank Rules

As a student
1. Go to the Peer Chat step and make sure the Question Bank shows up and the questions are clickable
2. Click on a question to add it to the student response input
3. Click "Send" to submit the response. The question in the question bank should show that it has been used.
4. Click on a question to add it to the student response input
5. Delete the question from the student response input. Type something else and submit the response. The question should not show that it has been used.
6. Leave the step
7. Go back to the Peer Chat step. The questions should show which ones have been used.

Make sure the old Peer Chat Question Bank still works as before for teacher and student.

Closes #1413
